jetson/nav/gate_search/gateStateMachine.cpp

```diff
@@ -218,74 +222,109 @@ NavState GateStateMachine::executeGateDriveToCentPoint()
-    const double fovAngle = mRoverConfig["computerVision"]["fieldOfViewSafeAngle"].GetDouble();
-    const Odometry currOdom = mPhoebe->roverStatus().odometry();
-
-    // If we are centered
-    const double targetAnglesDiff = mPhoebe->roverStatus().target().bearing +
-                                    mPhoebe->roverStatus().target2().bearing;
-    if(targetAnglesDiff < mRoverConfig["navThresholds"]["gateCenteredAngleDiff"].GetDouble())
+    if( mPhoebe->roverStatus().rightTarget().distance > 0 ) 
     {
-        direction = 1;
-        return NavState::GateDriveThrough;
+        if( mPhoebe->roverStatus().leftTarget().distance < mPhoebe->roverStatus().rightTarget().distance ) 
+        {
+            if( mPhoebe->turn(mPhoebe->roverStatus().rightTarget().bearing + mPhoebe->roverStatus().odometry().bearing_deg) ) 
+            {
+                return NavState::GateDriveToFarPost;
+            }
+        }
+        else 
+        {
+            if( mPhoebe->turn(mPhoebe->roverStatus().leftTarget().bearing + mPhoebe->roverStatus().odometry().bearing_deg) ) 
+            {
+                return NavState::GateDriveToFarPost;
+            }   
+        }
+    }
+    else 
+    {
+        if( mPhoebe->turn(mPhoebe->roverStatus().leftTarget().bearing + mPhoebe->roverStatus().odometry().bearing_deg) ) 
+        {
+            return NavState::GateDriveToFarPost;
+        }
+    }
+    return NavState::GateTurnToFarPost;
+} // executeGateTurnToFarPost()
+
+// Drive to furthest post (or the only post if only one is available)
+NavState GateStateMachine::executeGateDriveToFarPost()
+{
+    DriveStatus driveStatus;
+
+    // Set to first target, since we should have atleast one in sight/detected
+    double distance = mPhoebe->roverStatus().leftTarget().distance - gateAdjustmentDist;
+    double bearing = mPhoebe->roverStatus().leftTarget().bearing + mPhoebe->roverStatus().odometry().bearing_deg;
+
+    if( mPhoebe->roverStatus().rightTarget().distance > 0 ) 
+    {
+        if( mPhoebe->roverStatus().leftTarget().distance < mPhoebe->roverStatus().rightTarget().distance ) 
+        {
+            // Set our variables to drive to target/post 2, which is farther away
+            distance = mPhoebe->roverStatus().rightTarget().distance - gateAdjustmentDist;
+            bearing = mPhoebe->roverStatus().rightTarget().bearing + mPhoebe->roverStatus().odometry().bearing_deg;
+        }
     }
 
-    // If we need to switch directions
-    const bool visibleTargetAngles = mPhoebe->roverStatus().target().bearing > fovAngle / 2 &&
-                                     mPhoebe->roverStatus().target2().bearing < fovAngle / 2;
-    const bool visibleTargetDists = mPhoebe->roverStatus().target().distance < fovDepth &&
-                                    mPhoebe->roverStatus().target2().distance < fovDepth;
-    if(!visibleTargetAngles || !visibleTargetDists)
+    driveStatus = mPhoebe->drive( distance, bearing, true );
```

Comment:
Why is `driveStatus` declared above? What about `DriveStatus driveStatus = mPhoebe->...` and then remove the declaration above.